pkgs/development/tools/misc/instant-scala/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "instant-scala";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "jatcwang";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "jqSvKTL8NzqjwqDj/+55YWecx2bnzuArP8RdfH5q/1U=";
+  };
+  
+  propagateBuildInputs = [
```

Comment:
I have changed it to buildInputs for now but keen to learn which way is correct based on what this script is doing

Reply:
Well, first of all, it should be `propagatedBuildInputs`, so I don't believe this is doing anything but defining a list that is not used?

propagatedBuildInputs is only for buildInputs that have to be "on the PATH" during a build of something that depends on your derivation. Instead you should patch the script, with patchShebangs and/or manually so that that is not necessary.